File: src/main/java/com/sixtythreeebays/service/CartItemService.java
package com.sixtythreeebays.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.sixtythreeebays.domain.CartItem;
import com.sixtythreeebays.dto.CartItemDTO;
import com.sixtythreeebays.dto.CartItemSearchDTO;
import com.sixtythreeebays.dto.CartItemPageDTO;
import com.sixtythreeebays.dto.CartItemConvertCriteriaDTO;
import com.sixtythreeebays.service.GenericService;
import com.sixtythreeebays.dto.common.RequestDTO;
import com.sixtythreeebays.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CartItemService extends GenericService<CartItem, Integer> {

	List<CartItem> findAll();

	ResultDTO addCartItem(CartItemDTO cartItemDTO, RequestDTO requestDTO);

	ResultDTO updateCartItem(CartItemDTO cartItemDTO, RequestDTO requestDTO);

    Page<CartItem> getAllCartItems(Pageable pageable);

    Page<CartItem> getAllCartItems(Specification<CartItem> spec, Pageable pageable);

	ResponseEntity<CartItemPageDTO> getCartItems(CartItemSearchDTO cartItemSearchDTO);
	
	List<CartItemDTO> convertCartItemsToCartItemDTOs(List<CartItem> cartItems, CartItemConvertCriteriaDTO convertCriteria);

	CartItemDTO getCartItemDTOById(Integer cartItemId);







}

File: sixtythreeebays_ui/src/services/CartItemService.js
import http from "../http-common"; 

class CartItemService {
  getAllCartItems(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/cartItem/cartItems`, searchDTO);
  }

  get(cartItemId) {
    return this.getRequest(`/cartItem/${cartItemId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/cartItem?field=${matchData}`, null);
  }

  addCartItem(data) {
    return http.post("/cartItem/addCartItem", data);
  }

  update(data) {
  	return http.post("/cartItem/updateCartItem", data);
  }
  
  uploadImage(data,cartItemId) {
  	return http.postForm("/cartItem/uploadImage/"+cartItemId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CartItemService();
